bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/account/SmartherNotificationServlet.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smarther.internal.account;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.smarther.internal.api.dto.Notification;
+import org.openhab.binding.smarther.internal.api.exception.SmartherGatewayException;
+import org.openhab.binding.smarther.internal.api.exception.SmartherNotificationException;
+import org.openhab.binding.smarther.internal.util.ModelUtil;
+import org.openhab.binding.smarther.internal.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@code SmartherNotificationServlet} class acts as the registered endpoint to receive module status notifications
+ * from the Legrand/Bticino C2C Webhook notification service.
+ *
+ * @author Fabio Possieri - Initial contribution
+ */
+@NonNullByDefault
+public class SmartherNotificationServlet extends HttpServlet {
+
+    private static final long serialVersionUID = -2474355132186048438L;
+
+    private static final String CONTENT_TYPE = "application/json;charset=UTF-8";
+    private static final String OK_RESULT_MSG = "{\"result\":0}";
+
+    private final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final SmartherAccountService accountService;
+
+    /**
+     * Constructs a {@code SmartherNotificationServlet} associated to the given {@link SmartherAccountService} service.
+     *
+     * @param accountService
+     *            the account service to associate to the servlet
+     */
+    public SmartherNotificationServlet(SmartherAccountService accountService) {
+        this.accountService = accountService;
+    }
+
+    @Override
+    protected void doPost(@Nullable HttpServletRequest req, @Nullable HttpServletResponse resp)
+            throws ServletException, IOException {
+        if (req == null || resp == null) {
+            throw new SmartherNotificationException("Notification callback with null request/response");
```

Comment:
Indeed :) Changed to return a http 400 Bad Request in case of no request object, or throw a ServletException in case of no response object (as I cannot return any response in this case).
Same for `SmartherAuthorizationServlet`.